### PR TITLE
app deletion, subscription deletion workflow improvements

### PR DESCRIPTION
**Pull request by chameerar**
Introduces approval workflow flow for application deletion and subscription deletion

### CODE REVIEW

**Comment by chamilaadhi** [comment]
Reviewed comments can be found in https://github.com/wso2-support/carbon-apimgt/pull/5132

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
 import org.wso2.carbon.apimgt.api.model.SubscribedAPI;
 import org.wso2.carbon.apimgt.api.model.Subscriber;
 import org.wso2.carbon.apimgt.api.model.SubscriptionResponse;
+import org.wso2.carbon.apimgt.impl.*;
```
Comment (wasuradananjith) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java:40]: Why do we have this wildcard import here? Shall we remove it?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 2]
```diff
@@ -8441,6 +8441,42 @@ public String getExternalWorkflowReferenceByApplicationID(int appID) throws APIM
         return workflowExtRef;
     }
 
+    /**
+     * Get external workflow reference by internal workflow reference and workflow type
+     * @param internalRef Internal reference of the workflow
+     * @param workflowType Workflow type of the workflow
+     * @return External workflow reference for the given internal reference and workflow type if present. Null otherwise
+     * @throws APIManagementException
+     */
```
Comment (wasuradananjith) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:8450]: Javadoc comments are not properly formatted. Shall we fix those?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java (hunk 1 above)
Comment (chameerar) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java:40]: Fixed in - https://github.com/wso2/carbon-apimgt/pull/11699

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java (hunk 2 above)
Comment (chameerar) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:8450]: Fixed in - https://github.com/wso2/carbon-apimgt/pull/11699

**Pull request merged**